data/json/mapgen/mansion/mansion.json

```diff
@@ -0,0 +1,5841 @@
+      },
+      "place_monsters": [
+        { "monster": "GROUP_MANSION", "x": [ 2, 21 ], "y": [ 2, 7 ], "density": 0.08 }
+      ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [
+      "mansion_t_dn",
+      "mansion_t7d"
+    ],
+    "weight": 1000,
+    "object": {
+      "fill_ter": "t_rock_floor",
+      "rows": [
+        "X--CCCC--X----Xa(.KKdssX",
+        "---------X----X...KKdsKK",
+        "---------+----+....1.s.K",
+        "---------+----X.........",
+        "--TSSSST-X---NXtt.......",
+        "---------X<--NXtt.......",
+        "--SSTSSS-X<--FX.........",
+        "---------XXXXXXKK.......",
+        "--SSSTSS-X####XKK..///..",
+        "---------X####XX....KKK.",
+        "--SSTSST-X####X.1...KKK.",
+        "---------X####Xttt......",
+        "---------X####Xttt@q..vh",
+        "XXXXXXXXXX####XXXXXXXXXX",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################",
+        "########################"
+      ],
+      "terrain": {
+        "#": "t_rock",
+        "+": "t_door_c",
+        "-": "t_carpet_red",
+        "<": "t_stairs_up",
+        "C": "t_carpet_red",
+        "F": "t_carpet_red",
+        "N": "t_carpet_red",
+        "O": "t_window_domestic",
+        "S": "t_carpet_red",
+        "T": "t_carpet_red",
+        "X": "t_brick_wall"
+      },
+      "furniture": {
+        "(": "f_trashcan",
+        "/": "f_bench",
+        "1": "f_chair",
+        "@": "f_statue",
+        "C": "f_counter",
+        "F": "f_fridge",
+        "K": [ "f_crate_c", "f_crate_o" ],
+        "N": "f_cupboard",
+        "S": "f_sofa",
+        "T": "f_table",
+        "a": "f_armchair",
+        "d": "f_desk",
+        "h": "f_treadmill",
+        "q": "f_oven",
+        "t": "f_table"
+      },
+      "items": {
+        "(": { "item": "trash_cart", "chance": 30 },
+        "-": { "item": "clutter_mansion", "chance": 1 },
+        ".": { "item": "clutter_basement", "chance": 1 },
+        "C": { "item": "a_television", "chance": 100 },
+        "F": { "item": "fridgesnacks", "chance": 45 },
+        "K": { "item": "crate_stack", "chance": 100 },
+        "N": { "item": "vending_food", "chance": 40 },
+        "T": { "item": "snacks_fancy", "chance": 35 },
+        "d": { "item": "mansion_ammo", "chance": 40 }
+      },
+      "place_monsters": [
+        { "monster": "GROUP_MANSION", "x": [ 2, 21 ], "y": [ 2, 12 ], "density": 0.08 }
+      ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [
+      "mansion_t",
+      "mansion_t7"
+    ],
+    "weight": 1000,
+    "object": {
+      "fill_ter": "t_floor",
+      "rows": [
+        "X....nnn.X....X...
```

Comment:
This looks really wrong: `chance: 100` means only 1 in 100 safes will have anything in it.

Reply:
Is it? I thought 100 is for 100%.

Reply:
There were no issues in amount of loot when this mansion was in the mod. Will test it again later.

Reply:
Oh, I re-read it now. It looks like there are two different functions in mapgen and the item group one uses `x_in_y( chance, 100 )`, while `spawn_item` for some reason uses `one_in( chance )`. This is pretty terrible, but looks like the PR is not affected.